### PR TITLE
Update doctrine/orm 3.0

### PR DESCRIPTION
**Pull request by alioguz**

### CODE REVIEW

**Comment by gilles-g** [comment]
Hi @alioguz,

Thx for open the PR!

I have tried on my side to upgrade the bundle, but...
It's complicated, very.

In particular with the DBAL lib, they removed a lot of method.

Example:

The DBAL QueryBuilder cannot give us SQLParts, so...I think to use PHP Reflection to access on joins on the QueryBuilder

```php
public function hasJoinAlias($joinAlias): bool
    {
        $joinParts = $this->queryBuilder->getQueryPart('join');

        foreach ($joinParts as $rootAlias => $joins) {
            foreach ($joins as $join) {
                if ($join['joinAlias'] === $joinAlias) {
                    return true;
                }
            }
        }

        return false;
    }
```

Maybe @derrabus could give us an advice?

**Comment by Jean85** [comment]
@gilles-g keep in mind that you're not forced to upgrade to DBAL 4, you can explicitly lock against DBAL 3 and work it out from there, ORM 3 would still be installable. 

Still, a lot of people are NOT doing that, so you still need to find a way forward for that.

**Comment by Jean85** [comment]
> The DBAL QueryBuilder cannot give us SQLParts, so...I think to use PHP Reflection to access to "joins property" on the QueryBuilder

I delved a bit onto the repo, and I fear that the only way to proceed is to force the user to provide the root on which the filter has to be applied, basically requiring an `alias` property on all `*FilterType` classes.

**Comment by gilles-g** [comment]
@alioguz Could you try to rebase? 
I have add a github workflow to check BC :D 

Thx for your work 👍

File: composer.json [hunk 1]
```diff
@@ -7,17 +7,18 @@
     "license": "MIT",
     "prefer-stable": true,
     "require": {
-        "php": ">=7.4",
-        "doctrine/orm": "^2.10",
-        "symfony/form": "^5.3|^6.0|^7.0",
-        "symfony/framework-bundle": "^5.3|^6.0|^7.0"
+        "php": "^8.1",
+        "doctrine/orm": "^3.0",
```
Comment (gilles-g) [inline composer.json:11]: @alioguz Do you try to allow  orm 2 and dbal 2 ?

Comment (alioguz) [reply composer.json:11]: Yes, I tried this config : 

`
"php": "^8.1",
"doctrine/orm": "^2.0",
"doctrine/dbal": "^2.0",
`

But an error occurred when I launch `composer test` : 

`PHP Fatal error:  Declaration of Doctrine\Bundle\DoctrineBundle\DependencyInjection\DoctrineExtension::getMappingResourceConfigDirectory(): string must be compatible with Symfony\Bridge\Doctrine\DependencyInjection\AbstractDoctrineExtension::getMappingResourceConfigDirectory(?string $bundleDir = null): string in /home/ali/dev/form-filter-bundle/vendor/doctrine/doctrine-bundle/DependencyInjection/DoctrineExtension.php on line 868`

**Comment by alioguz** [comment]
@gilles-g Can you review please ?

**Pull request merged**